src/core/http/api/AuthApi.ts

```diff
@@ -0,0 +1,72 @@
+import { BaseApi } from "./BaseApi";
+import { HTTPMethod } from "./types/HTTPTypes";
+import { RequestResult } from "./types/RequestResult";
+import { SwaggerUserInfo } from "./types/SwaggerTypes";
+
+export class AuthApi extends BaseApi {
+  /**
+   * Регистрация
+   * @param data Регистрационные данные
+   */
+  signUp(data: {
+    first_name: string,
+    second_name: string,
+    login: string,
+    email: string,
+    password: string,
+    phone: string
+  }
+  ) : Promise<{id: number}> {
```

Comment:
Стоит смержить main сюда и прогнать через prettier, потому что он вот только добавился в проект.
Так же возвращаемый тип функций стоит либо писать везде, либо не писать нигде
```suggestion
  ): Promise<{id: number}> {
```